src/error_gui.cpp

```diff
@@ -137,14 +120,9 @@ void ErrorMessageData::CopyOutDParams()
 		if (company < MAX_COMPANIES) face = company;
 	}
 
-	/* Reset parameters */
-	for (size_t i = 0; i < lengthof(this->strings); i++) free(this->strings[i]);
-	memset(this->decode_params, 0, sizeof(this->decode_params));
-	memset(this->strings, 0, sizeof(this->strings));
-
 	/* Get parameters using type information */
 	if (this->textref_stack_size > 0) StartTextRefStackUsage(this->textref_stack_grffile, this->textref_stack_size, this->textref_stack);
-	CopyOutDParam(this->decode_params, this->strings, this->detailed_msg == INVALID_STRING_ID ? this->summary_msg : this->detailed_msg, lengthof(this->decode_params));
+	CopyOutDParam(this->params, 20, this->detailed_msg == INVALID_STRING_ID ? this->summary_msg : this->detailed_msg);
```

Comment:
Magic 20, presumably that's just some limit that was set elsewhere before.

Reply:
Yes, error.h:34/355 to be precise. In the end only the needed amount of parameters should be copied, but that's for later.